llvm-passes/ObfuscateVar/ObfuscateVar.cpp

```diff
@@ -0,0 +1,210 @@
+//Compile : cmake -DLLVM_ROOT=$HOME/Documents/Programmation/Obfuscation/llvm/build ..
+#include "llvm/Pass.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/Transforms/Utils/BasicBlockUtils.h"
+#include "llvm/Transforms/IPO/PassManagerBuilder.h"
+
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/IRBuilder.h"
+
+#ifndef NDEBUG
+#include "llvm/IR/Verifier.h"
+#include "llvm/Support/Debug.h"
+#endif
+
+#include <map>
+#include <utility> //std::pair, std::make_pair
+#include <random>
+
+
+using namespace llvm;
+
+namespace {
+  typedef uint32_t typeInter;
+  typedef std::pair<AllocaInst*,AllocaInst*> typeMapValue;
+  typedef std::map<Value*,typeMapValue> typeMap;
```

Comment:
LLVM provides the ``ValueMap`` type for this kind of mapping, see http://llvm.org/docs/doxygen/html/ValueMap_8h_source.html